botensure.go

```diff
@@ -0,0 +1,172 @@
+package pluginapi
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/mattermost/mattermost-server/v5/utils"
+	"github.com/pkg/errors"
+)
+
+type ensureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*ensureBotOptions)
+
+func (b *BotService) ensureServerVersion(required string) error {
```

Comment:
As an unrelated aside, I think we should eventually lift this out into a top-level service/function, and cache the current server version for the lifetime of the process.

Reply:
Made a ticket: https://mattermost.atlassian.net/browse/MM-23500